services/Skyra.Database/Extensions/DbSetExtensions.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace Skyra.Database.Extensions
+{
+	public static class DbSetExtensions
+	{
+		public async static Task<T> UpsertAsync<T>(this DbSet<T> db, object key, Func<T> data) where T : class
+		{
+			var entity = await db.FindAsync(key);
+
+			if (entity is null)
+			{
+				entity = data();
+				await db.AddAsync(entity);
+			}
```

Comment:
```suggestion
			entity ??= data();
			await db.AddAsync(entity);
```

Reply:
I don't think it's wise to always force an `AddAsync`.

Reply:
Good point!